src/BuildScriptGenerator/BaseBashBuildScript.sh.tpl

```diff
@@ -25,7 +25,7 @@ then
     DESTINATION_DIR=$(pwd -P)
 fi
 
-if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ]
+if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ] && [ "$CLEAN_DESTINATION_DIR" == "true" ]
```

Comment:
@gpcastro What do you think about this? I think we should not have been doing this even for Node and Python because @JennyLawrance would be using an intermediate directory for building and so this condition would have been true and we would be deleting the contents of the output directory there by deleting any persistent data that was in the output folder.

Reply:
Right. I'd make this a flag though, instead of environment variable, and default to not cleaning it as we dont know what might be in the output directory

Reply:
Removed this piece of code as per our discussion offline. Users can clean up if they want to using a pre-build script.